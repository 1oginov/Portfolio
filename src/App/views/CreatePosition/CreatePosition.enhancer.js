/* @flow */

import { compose, withHandlers, type HOC } from 'recompose';

import withNavigationHandlers from '../../enhancers/withNavigationHandlers';
import * as R from '../../routes';

const enhancer: HOC<*, *> = compose(
  withNavigationHandlers({
    handleHomeClick: R.HOME,
  }),
  withHandlers({

    handleOnCreate: ({ handleHomeClick }) => () => {
      handleHomeClick();
    },

  }),
);

export default enhancer;
