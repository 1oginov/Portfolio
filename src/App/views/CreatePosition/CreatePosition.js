/* @flow */

import * as React from 'react';

const CreatePosition = ({ handleHomeClick }) => (
  <React.Fragment>

    <div>
      <button onClick={handleHomeClick}>Home</button>
    </div>

    <h1>Create position</h1>

  </React.Fragment>
);

export default CreatePosition;
