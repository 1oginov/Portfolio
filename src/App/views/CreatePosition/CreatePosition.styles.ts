import { withStyles } from '@material-ui/core';

export default withStyles(theme => ({
  headline: {
    marginBottom: theme.spacing(3),
    marginTop: theme.spacing(3),
    textAlign: 'center',
  },
  root: {
    alignItems: 'center',
    display: 'flex',
    flexDirection: 'column',
    padding: theme.spacing(6),
  },
}));
