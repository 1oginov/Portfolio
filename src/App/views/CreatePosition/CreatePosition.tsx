import * as React from 'react';

import CreatePositionForm from 'Portfolio/components/OpenPositionForm';

export interface Props {
  handleCreate: () => void;
}

const CreatePosition: React.FunctionComponent<Props> = ({ handleCreate }: Props) => (
  <React.Fragment>

    <h1>Create position</h1>

    <CreatePositionForm onCreate={handleCreate} />

  </React.Fragment>
);

export default CreatePosition;
