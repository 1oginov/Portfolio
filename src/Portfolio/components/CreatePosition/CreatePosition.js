/* @flow */

import * as React from 'react';

const CreatePosition = ({
  amount, date, handleAmountChange, handleDateChange, handlePriceChange,
  handleSymbolChange, handleSubmit, price, symbol,
}) => (
  <form onSubmit={handleSubmit}>

    <input onChange={handleSymbolChange} type="text" value={symbol} />

    <input
      min="0"
      onChange={handlePriceChange}
      step="0.01"
      type="number"
      value={price}
    />

    <input min="1" onChange={handleAmountChange} type="number" value={amount} />

    <input onChange={handleDateChange} type="date" value={date} />

    <button type="submit">Create</button>

  </form>
);

export default CreatePosition;
