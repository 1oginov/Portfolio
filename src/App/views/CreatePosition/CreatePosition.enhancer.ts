import { RouteComponentProps as WithRouterProps, withRouter } from 'react-router-dom';
import { compose, withHandlers } from 'recompose';

import { Position } from 'Portfolio/lib';

import { Props } from './CreatePosition';
import * as R from '../../routes';

export default compose<Props, {}>(
  withRouter,
  withHandlers<WithRouterProps, {}>({

    handleCreate: ({ history }) => (position: Position) => {
      history.push(R.toPosition(position.id));
    },

  }),
);
