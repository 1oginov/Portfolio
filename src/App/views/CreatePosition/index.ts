import CreatePosition from './CreatePosition';
import enhance from './CreatePosition.enhancer';

export default enhance(CreatePosition);
