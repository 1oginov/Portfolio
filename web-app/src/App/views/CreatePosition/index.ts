import CreatePosition from './CreatePosition';
import enhance from './CreatePosition.enhancer';
import style from './CreatePosition.styles';

export default style(enhance(CreatePosition));
