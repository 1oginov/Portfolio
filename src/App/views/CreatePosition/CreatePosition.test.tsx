import { shallow } from 'enzyme';
import classes from 'identity-obj-proxy';
import * as React from 'react';

import CreatePosition from './CreatePosition';

it('matches snapshot', () => {
  const wrapper = shallow(
    <CreatePosition classes={classes} handleCreate={() => undefined} />,
  );

  expect(wrapper).toMatchSnapshot();
});
