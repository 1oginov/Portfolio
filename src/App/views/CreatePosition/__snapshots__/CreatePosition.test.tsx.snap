// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<Fragment>
  <h1>
    Create position
  </h1>
  <Connect(withStateHandlers(withHandlers(OpenPositionForm)))
    onCreate={[Function]}
  />
</Fragment>
`;
